### PR TITLE
Redesign cth macro

### PR DESCRIPTION
**Pull request by newfrenchy83**
I add cth ability with object for revert modifications was more easy

### CODE REVIEW

File: data/core/macros/utils.cfg [hunk 1]
```diff
@@ -534,50 +534,29 @@
...
-
-        [unstore_unit]
-            variable=unit
-            find_vacant=no
-        [/unstore_unit]
+        [object]
+		id=forced_cth
+		[filter]
+			{FILTER}
+		[/filter]
```
Comment (jostephd) [inline data/core/macros/utils.cfg]: [object][filter] gives the object only to the _first_ unit matching the filter, so your change breaks the macro in the case that {FILTER} matches more than one unit.

File: data/core/macros/utils.cfg [hunk 2]
```diff
@@ -685,26 +630,35 @@
...
-
-                    {CLEAR_VARIABLE this_item.specials.original_chance_to_hit}
-                [/do]
-            [/foreach]
-
-            [unstore_unit]
-                variable=second_unit
-                find_vacant=no
-            [/unstore_unit]
+             [remove_object]
```
Comment (jostephd) [inline data/core/macros/utils.cfg]: Thank you for fixing the indentation above. The indentation from this point to the end is wrong.

**Comment by newfrenchy83** [comment]
I close this PR and i will commit anorher.

**Comment by jostephd** [comment]
When you do that please link the new PR and this PR to each other.

**Pull request closed without merging**